frontends/verilog/verilog_parser.y

```diff
@@ -355,11 +369,12 @@ hierarchical_id:
 
 hierarchical_type_id:
 	TOK_USER_TYPE
+	| TOK_PKG_USER_TYPE				// package qualified type name
 	| '(' TOK_USER_TYPE ')'	{ $$ = $2; }		// non-standard grammar
 	;
 
 module:
-	attr TOK_MODULE TOK_ID {
+	attr module_start TOK_ID {
```

Comment:
Why not simply run `enterTypeScope();` right here? (After all, we also have `exitTypeScope();` in this production.) Something like

```
module:
        attr TOK_MODULE {
                enterTypeScope();
        } TOK_ID {
                do_not_require_port_stubs = false;
                AstNode *mod = new AstNode(AST_MODULE);
                ast_stack.back()->children.push_back(mod);
                ast_stack.push_back(mod);
                current_ast_mod = mod;
                port_stubs.clear();
                port_counter = 0;
                mod->str = *$4;
                append_attr(mod, $1);
                delete $4;
        } module_para_opt module_args_opt ';' module_body TOK_ENDMODULE {
                if (port_stubs.size() != 0)
                        frontend_verilog_yyerror("Missing details for module port `%s'.",
                                        port_stubs.begin()->first.c_str());
                SET_AST_NODE_LOC(ast_stack.back(), @2, @$);
                ast_stack.pop_back();
                log_assert(ast_stack.size() == 1);
                current_ast_mod = NULL;
                exitTypeScope();
        };
```